pulsar-broker/src/main/java/org/apache/pulsar/broker/rest/Topics.java

```diff
@@ -0,0 +1,133 @@
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public void produceOnNonPersistentTopic(@Suspended final AsyncResponse asyncResponse,
+                                         @ApiParam(value = "Specify the tenant", required = true)
+                                         @PathParam("tenant") String tenant,
+                                         @ApiParam(value = "Specify the namespace", required = true)
+                                         @PathParam("namespace") String namespace,
+                                         @ApiParam(value = "Specify topic name", required = true)
+                                         @PathParam("topic") @Encoded String encodedTopic,
+                                         @QueryParam("authoritative") @DefaultValue("false")
+                                                        boolean authoritative,
+                                         ProducerMessages producerMessages) {
+        validateTopicName(tenant, namespace, encodedTopic);
+        publishMessages(asyncResponse, producerMessages, authoritative);
+    }
+
+    @POST
+    @Path("/non-persistent/{tenant}/{namespace}/{topic}/partitions/{partition}")
+    @ApiOperation(value = "Produce message to a partition of a persistent topic.",
+            response = String.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "tenant/namespace/topic doesn't exit"),
+            @ApiResponse(code = 412, message = "Namespace name is not valid"),
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public void produceOnNonPersistentTopicPartition(@Suspended final AsyncResponse asyncResponse,
+                                                  @ApiParam(value = "Specify the tenant", required = true)
+                                                  @PathParam("tenant") String tenant,
+                                                  @ApiParam(value = "Specify the namespace", required = true)
+                                                  @PathParam("namespace") String namespace,
+                                                  @ApiParam(value = "Specify topic name", required = true)
+                                                  @PathParam("topic") @Encoded String encodedTopic,
+                                                  @ApiParam(value = "Specify topic partition", required = true)
+                                                  @PathParam("partition") int partition,
+                                                  @QueryParam("authoritative") @DefaultValue("false")
+                                                                 boolean authoritative,
+                                                  ProducerMessages producerMessages) {
+        validateTopicName(tenant, namespace, encodedTopic);
+        publishMessagesToPartition(asyncResponse, producerMessages, authoritative, partition);
```

Comment:
There seems to be a security issue, can I send messages to any topic?

Reply:
yeah, we're missing permission check here, 
added check to use AuthorizationService#canProduce to verify caller have the permission to publish on given topic.